app/src/components/InventorySelect.tsx

```diff
@@ -33,7 +33,9 @@ export const InventorySelect = ({
     await setUserInfo({
       cityId: inventory.cityId!,
       defaultInventoryId: inventory.inventoryId,
-    }).unwrap();
+    }).then(() => {
+      router.push(`/${inventory.inventoryId}`);
+    });
```

Comment:
Don't remove the unwrap as that means there is no promise AFAIK.
Also since this is an async function, you don't need to use .then but you can just put the router.push line after the await.